Improving onboarding (#10469)
solo-io/gloo: docs/content/guides/dev/setting-up-dev-environment.md

Developing on Gloo Gateway requires the following to be installed on your system:

- [`Make`](https://www.gnu.org/software/make/)
- [`Git`](https://git-scm.com/)
- [`Go`](https://golang.org/) (`solo-io` projects are built using version `1.23.3`)
- `Protoc` (`solo-io` projects are built using version `3.6.1`)
- Standard development tools like `gcc`

To install all the requirements, do the following:
```bash
# install build tools
# - macOS
xcode-select --install
# - other operating systems:
# distro meta-packages, like "build-essential", should have what is required

############################################
# install version of go in go.mod
############################################
# - macOS:
brew install go@1.23
# - other operating systems:
# follow directions at https://go.dev/doc/install

# install protoc
make install-protoc

# install go related tools
make install-go-tools
```

> ### Setting up the Gloo Gateway Repositories

Reviewer: ```suggestion
### Setting up the Gloo Gateway repositories
```

go run projects/discovery/cmd/main.go
# run gateway locally
go run projects/gateway/cmd/main.go
```

Awesome! You're ready to start developing on Gloo Gateway! Check out the [Writing Upstream Plugins Guide]({{% versioned_link_path fromRoot="/guides/dev/writing-upstream-plugins" %}}) to see how to add plugins to gloo.

### Developing with a local K8s cluster (kind)

> Developers without a K8s cluster are encouraged to use [Docker](https://docs.docker.com) and [Kind](https://kind.sigs.k8s.io).

Reviewer: ```suggestion
Developers without a K8s cluster are encouraged to use [Docker](https://docs.docker.com) and [Kind](https://kind.sigs.k8s.io) to build a development and test environment. 
```

> Prerequisites:

Reviewer: ```suggestion
Prerequisites:
Install the following tools: 
```

* [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
* [Docker](https://docs.docker.com/install/)
* [Helm](https://github.com/helm/helm)

> Stand up Gloo Gateway running on Kind and begin making changes:

Reviewer: ```suggestion
Install Gloo Gateway on a Kind cluster so that you can start making changes to the Gloo Gateway code. 
```

```bash

> # prepare kind cluster, build images, and upload them

Reviewer: ```suggestion
# Prepare the kind cluster, build images, and upload them.
```

make kind-setup

> # install Gloo

Reviewer: ```suggestion
# install Gloo Gateway
```

helm upgrade --install -n gloo-system --create-namespace gloo ./_test/gloo-1.0.1-dev.tgz --values ./test/kubernetes/e2e/tests/manifests/common-recommendations.yaml

############################################
# make changes to the code in the repo ...
############################################

> # update what is running in the cluster with your changes

Reviewer: ```suggestion
# Update the Gloo Gateway installation in your cluster to apply your changes
```

make -B kind-build-and-load-gloo
```

### Code Generation

> Follow these steps if you are making changes to Gloo Gateway's Protobuf-based API.

Reviewer: ```suggestion
Follow these steps to make changes to Gloo Gateway's Protobuf-based API.
```
